src/main/java/org/spongepowered/common/effect/particle/SpongeParticleHelper.java

```diff
@@ -111,12 +112,23 @@ private static int getDirectionData(Direction direction) {
         }
     }
 
-    private static final EntityFireworkRocket FIREWORK_ROCKET_DUMMY = new EntityFireworkRocket(null);
-    private static final Packet<?> DESTROY_FIREWORK_ROCKET_DUMMY = new SPacketDestroyEntities(FIREWORK_ROCKET_DUMMY.getEntityId());
-    private static final Packet<?> FIREWORK_ROCKET_DUMMY_EFFECT = new SPacketEntityStatus(FIREWORK_ROCKET_DUMMY, (byte) 17);
+    // Get the next free entity id
+    private static final int FIREWORK_ROCKET_ID;
+    private static final UUID FIREWORK_ROCKET_UNIQUE_ID;
+
+    private static final SPacketDestroyEntities DESTROY_FIREWORK_ROCKET_DUMMY;
+    private static final SPacketEntityStatus FIREWORK_ROCKET_DUMMY_EFFECT;
 
     static {
-        FIREWORK_ROCKET_DUMMY.getDataManager().getDirty();
+        final EntityFireworkRocket entityFireworkRocket = new EntityFireworkRocket(null);
+        FIREWORK_ROCKET_ID = entityFireworkRocket.getEntityId();
+        FIREWORK_ROCKET_UNIQUE_ID = entityFireworkRocket.getUniqueID();
+
+        DESTROY_FIREWORK_ROCKET_DUMMY = new SPacketDestroyEntities(FIREWORK_ROCKET_ID);
+
+        FIREWORK_ROCKET_DUMMY_EFFECT = new SPacketEntityStatus();
+        FIREWORK_ROCKET_DUMMY_EFFECT.entityId = FIREWORK_ROCKET_ID;
+        FIREWORK_ROCKET_DUMMY_EFFECT.logicOpcode = 17;
```

Comment:
Where does this number come from? A comment on this line would be nice for anyone looking at this in the future.